utensor_cgen/legalizer/onnx.py

```diff
@@ -0,0 +1,57 @@
+from utensor_cgen.ir import OperationInfo, TensorInfo
+from utensor_cgen.utils import topologic_order_graph
+
+from .base import LegalizerBase
+
+
+class OnnxLegalizer(LegalizerBase):
+  TARGET = 'onnx'
+
+  def legalize(self, ugraph):
+    self._visit(ugraph)
+    return ugraph
+
+  def _visit(self, ugraph):
+    for op_info in list(ugraph.ops_info.values()):
+      visitor = getattr(
+        self,
+        '_visit_{}'.format(op_info.op_type.lower()),
+        lambda op_info: op_info
+      )
+      visitor(op_info)
+    topologic_order_graph(ugraph)
+
+  def _visit_gemm(self, op_info):
```

Comment:
Where do these functions get called from, I noticed this pattern used elsewhere but it's a bit unclear how they're used

Reply:
https://github.com/uTensor/utensor_cgen/pull/102/files/7d9d6eaeb9e380d233580612cb173e2ea7085e77#diff-78a1f0eaee7e58baa9a6b8b7c7954f74R16-R21

I use dynamic attribute lookup in python.
The `getattr` function.

Reply:
Might consider adding this as a mini comment above, would be super helpful

Reply:
Ok

Reply:
Done

Reply:
<3